### PR TITLE
Fix: Prevent page reloads twice when the shopper sets a price filter and attribute filter using blocks in a PHP template

### LINKED ISSUE #6330: Page reloads twice when the shopper sets a price filter and attribute filter using blocks in a PHP template.
When Filter Products by Price and Filter Products by Attribute are used in the Product Catalog template and the shopper selects both filters, the page reloads twice.

## To reproduce

1. With a block theme, go to Appearance > Site Editor.
2. Go to Templates and edit the Product Catalog template.
3. Above the WooCommerce Classic Template block, add the Filter Products by Price and Filter Products by Attribute blocks.
4. In the frontend, filter by price and let the page reload.
5. Now, filter also by attribute.
6. Notice the page reloads twice.

## Expected behavior

The page should only reload once.

## Screenshots

https://user-images.githubusercontent.com/3616980/165425137-34a877d9-f74f-436e-9a2a-7a8131817727.mp4

### PR DESCRIPTION
**Pull request by dinhtungdu**
Fixes #6330 

This PR fixes the issue when the page reloads twice after setting a price filter and attribute filter using blocks in a PHP template. The root of the issue is that we're tracking the change in the state and updating the price filter block when the state changes. This works for blocks but doesn't work well for PHP templates where the state is shared between page loads through the URL causing the additional page reload (thanks @tjcafferkey for [the investigation](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6330#issuecomment-1111992728)).

One more thing I noticed that we may want to fix: When using with products grid block like All Products, when the shopper set a price filter then an attribute filter, the product grid flashes because of the same reason pointed out above. The product grid first loads the result, then flashes, and then shows the same result, please check [this screencast](https://user-images.githubusercontent.com/5423135/166095061-5d1ad527-e931-4af0-af3f-58f2e7ea0185.mov) for more info.

### Screencasts

<table>
<tr>
	<td> Before
	<td> After
<tr>
	<td> <video src="https://user-images.githubusercontent.com/3616980/165425137-34a877d9-f74f-436e-9a2a-7a8131817727.mp4">
	<td> <video src="https://user-images.githubusercontent.com/5423135/166094595-b5a3cac1-2e59-44bb-be3a-3f069739f91d.mov">
</table>

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. With a block theme, go to Appearance > Site Editor.
2. Go to Templates and edit the Product Catalog template.
3. Above the WooCommerce Classic Template block, add the Filter Products by Price and Filter Products by Attribute blocks.
4. In the frontend, filter by price and let the page reload.
5. Now, filter also by attribute.
6. Notice the page **not** reloads twice.

* [ ] Do not include in the Testing Notes

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: assets/js/blocks/price-filter/block.js [hunk 1]
```diff
@@ -202,20 +202,38 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 			if ( prices[ 1 ] !== maxPrice ) {
 				setMaxPrice( prices[ 1 ] );
 			}
+
+			if (
+				filteringForPhpTemplate &&
+				hasSetPhpFilterDefaults &&
+				! attributes.showFilterButton
+			) {
+				onSubmit( prices[ 0 ], prices[ 1 ] );
```
Comment (Aljullu) [inline assets/js/blocks/price-filter/block.js]: With this solution, moving the slider from `0` to `70` will cause 7 requests (`10`, `20`, `30`, ...), no? (At least, that's what I noticed in my testing)

Should we create a wrapper around `onSubmit` to debounce it?

**Review by Aljullu — COMMENTED** [review]
Thanks for investigating this and identifying the source of the issue @dinhtungdu! The current solution looks good to me and seems to fix the issue, but I left a question below regarding whether we should debounce the `onSubmit`.

File: assets/js/blocks/price-filter/block.js (hunk 1 above)
Comment (dinhtungdu) [reply assets/js/blocks/price-filter/block.js]: Thanks so much for this! It's my bad that I didn't test the block without the filter button. I addressed your review and this PR is ready to review again!

**Review by tjcafferkey — APPROVED** [review]
Thanks for this @dinhtungdu and for addressing the feedback by @Aljullu I can confirm this works as expected and fixes the issue 🙌🏻

**Comment by dinhtungdu** [comment]
Failed E2E test is unrelated. Merging with failed workflows.

**Pull request merged**